packages/eslint-plugin/docs/rules/restrict-plus-operands.md

```diff
@@ -18,70 +18,159 @@ This rule reports when a `+` operation combines two values of different types, o
 ### ❌ Incorrect
 
 ```ts
-var foo = '5.5' + 5;
-var foo = 1n + 1;
+let foo = '5.5' + 5;
+let foo = 1n + 1;
 ```
 
 ### ✅ Correct
 
 ```ts
-var foo = parseInt('5.5', 10) + 10;
-var foo = 1n + 1n;
+let foo = parseInt('5.5', 10) + 10;
+let foo = 1n + 1n;
 ```
 
 ## Options
 
-### `checkCompoundAssignments`
+:::caution
+We generally recommend against using these options, as they limit which varieties of incorrect `+` usage can be checked.
 
-Examples of code for this rule with `{ checkCompoundAssignments: true }`:
+For example, instead of enabling the `allow*` options to concatenate strings and other values, use APIs akin to `console.log` that allow passing multiple values:
+
+```diff
+declare const value: boolean;
+
+- console.log("My value is: " + value);
++ console.log("My value is:", value;
+```
+
+:::
```

Comment:
@bradzacher does this look like enough cautioning? Think there's anything else we should do to yell at people not to use these things?